models/article-model.js

```diff
@@ -1,13 +1,39 @@
-const db = require('../db/connection')
+const db = require("../db/connection");
 
-exports.fetchArticleByID = (articleID) =>{
-    return db.query(`SELECT * FROM articles WHERE article_id =${articleID};`).then(({rows})=>{
-        if(rows.length===0){
-            return Promise.reject({status: 400, message: "invalid query"})
-            }
-            return rows[0]
- } ).catch((err)=>{
-throw err
- })
-     }
-    
\ No newline at end of file
+exports.fetchArticleByID = (articleID) => {
+  return db
+    .query(`SELECT *FROM articles WHERE article_id =${articleID}`)
+    .then(({ rows }) => {
+        const article = rows[0]
+      if (!article) {
+        return Promise.reject({ status: 404, msg: "Not found" });
+      }
+      return article
+    })
+    .catch((err) => {
+        throw err;
+      });
+};
+
+
+exports.fetchAllArticles = () =>{
+    return db
+    .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id =articles.article_id
+        GROUP BY 
+        articles.article_id, 
+        articles.title, 
+        articles.topic, 
+        articles.author, 
+        articles.created_at, 
+        articles.votes, 
+        articles.article_img_url
+      ORDER BY articles.created_at DESC;
+    `)
+    .then(({ rows }) => {
+      return rows;
+    })
+    .catch((err) => {
+      console.error(err + '  <<< error in model')
+      return Promise.reject({ status: 500, message: "Internal Server Error" }); 
+    });
```

Comment:
You don't need an error test for 500. PSQL can handle those errors for us quite nicely 👍